pkg/provider/pods.go

```diff
@@ -447,28 +467,97 @@ func (p *Pod) IsUsingSRIOVWithMTU() (bool, error) {
 
 	// For each CNCF network, get its network attachment definition and check
 	// whether its config's type is "sriov"
+
 	oc := clientsholder.GetClientsHolder()
 
+	// Steps:
+	// 1. Compare the network name with the NAD name and check if the MTU is set.
+	// 2. If the MTU is not set in the NAD config, we should double-check the network-status annotation.
+	//    The network status (if the NAD name matches) could possibly have the MTU set.
+	// 3. If neither of the above steps is true, then check the SriovNetwork/SriovNetworkNodePolicy CRs
+
 	for _, networkName := range cncfNetworkNames {
 		log.Debug("%s: Reviewing network-attachment definition %q", p, networkName)
-		nad, err := oc.CNCFNetworkingClient.NetworkAttachmentDefinitions(
+		nad, err := oc.CNCFNetworkingClient.K8sCniCncfIoV1().NetworkAttachmentDefinitions(
 			p.Namespace).Get(context.TODO(), networkName, metav1.GetOptions{})
 		if err != nil {
 			return false, fmt.Errorf("failed to get NetworkAttachment %s: %v", networkName, err)
 		}
 
-		isSRIOV, err := isNetworkAttachmentDefinitionSRIOVConfigMTUSet(nad.Spec.Config)
+		// Check the NAD config to see if the MTU is set
+		isSRIOVwithMTU, err := isNetworkAttachmentDefinitionSRIOVConfigMTUSet(nad.Spec.Config)
 		if err != nil {
-			return false, fmt.Errorf("failed to know if network-attachment %s is sriov with MTU: %v",
-				networkName, err)
+			log.Error("Failed to know if network-attachment %s is sriov with MTU: %v", networkName, err)
+			return false, nil
 		}
 
 		log.Debug("%s: NAD config: %s", p, nad.Spec.Config)
-		if isSRIOV {
+		if isSRIOVwithMTU {
 			return true, nil
 		}
+		// If the NAD is defined and the MTU value is not found, let's check
+		// the network-status annotation to see if the MTU is set there and matches
+		// the NAD name.
+
+		// Get the network-status annotation (if any)
+		if networkStatuses, exist := p.Annotations[CniNetworksStatusKey]; exist {
+			networkStatusResult, err := networkStatusUsesMTU(networkStatuses, nad.Name)
+			if err != nil {
+				log.Error("Failed to know if network-status %s is sriov with MTU: %v", networkName, err)
+				return false, nil
```

Comment:
same, I'd not return anything here and keep checking, we already have the return at the end of the function with the false result

Reply:
Done.